woocommerce/woocommerce-ios
# App Store Listings: Update scripts for screenshots

### Description

As per request from the marketing team (p91TBi-9eM-p2), this PR updates the automation scripts to take requested screenshots.

 **Capturing the collect payment alert**:
- Hardware: we need to create a mock card reader and card reader status, so public initializers for these had to be added.
- UI Layer: additional accessibility identifiers to access the buttons and detect the collect payment modal.
- Woo Foundation: a minor update to `CurrencyFormatter` was added to transform value from non-Latin to Latin. This is necessary to format the order total value from Arabic numerals to Latin ones. `CollectOrderPaymentUseCase` needs to check this value for collecting payments.
- Yosmite: add mocks for `OrderCardPresentPaymentEligibilityAction`, `SystemStatusAction` and `CardPresentPaymentAction`.

**Simulating push notifications**
We want to capture push notifications on the app, so the trick is to use the local notification API:
- Added a new process argument and requested permission for notification upon launch when this argument is enabled.
- To capture the screenshot for the notification, lock the device screen and sleep for 2 seconds before taking a screenshot. In the meantime, schedule a local notification in 1 second when the app is about to be sent to the background. This way the lock screen can be captured with the notification coming in.

**Capturing the monthly revenue**
Marketing needs to switch to capture the This Month tab of the My Store screen, so I added mock data for the monthly stats.

### Testing instructions

- Turn off the feature flag `.splitViewInOrdersTab` in `DefaultFeatureFlagService`.
- Optionally set the simulator to a non-English language to make sure the test isn’t relying on anything English-specific. Try Arabic to check the case with Arabic numerals.
- Select target WooCommerceScreenshots. Run the `testScreenshots()` UI test.
- Confirm that the test succeeds.

### Screenshots

N/A

---
